spec/posthog/client_spec.rb

```diff
@@ -419,6 +419,42 @@ class PostHog
         properties = client.dequeue_last_message[:properties]
         expect(properties['$groups']).to eq({ 'company' => 'id:5', 'instance' => 'app.posthog.com' })
       end
+
+      it 'captures uuid when provided' do
+        client.capture(
+          {
+            distinct_id: 'distinct_id',
+            event: 'test_event',
+            uuid: '123e4567-e89b-12d3-a456-426614174000'
+          }
+        )
+        last_message = client.dequeue_last_message
+        expect(last_message['uuid']).to eq('123e4567-e89b-12d3-a456-426614174000')
+      end
+
+      it 'generates uuid when not provided' do
```

Comment:
Confusing test name. The UUID is not generated by the gem, but rather by the ingestion pipeline

Reply:
```suggestion
      it 'does not require a uuid be provided - ingestion will generate when absent' do
```